### PR TITLE
sql: some more postgres compat for FDW support

### LINKED ISSUE #20271: sql: support no-op WORK/TRANSACTION modifiers on ABORT
See https://www.postgresql.org/docs/8.2/static/sql-abort.html

Some clients apparently like to put these no-op modifiers on abort, we should support them. Needed by #20249.

### PR DESCRIPTION
**Pull request by jordanlewis**
- support ABORT [TRANSACTION | WORK]
- support INTERVALSTYLE

Updates #20249.
Closes #20271.

### CODE REVIEW

**Comment by knz** [comment]
:+1: on the compat change!

:-1: on exposing the ABORT syntax in docs

Also see #20264 for comments on the new session var

---

Reviewed 3 of 3 files at r1, 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/sql/parser/sql.y, line 3594 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/20274#-KzyR-rFsbitDG3OroDv:-KzyR-rFsbitDG3OroDw:bdvvs6d) ([raw file](https://github.com/cockroachdb/cockroach/blob/6cff7e81486c04f9b6689f8ce786c825ebafebaf/pkg/sql/parser/sql.y#L3594)):*
> ```Yacc
> | commit_stmt   // EXTEND WITH HELP: COMMIT
> | rollback_stmt // EXTEND WITH HELP: ROLLBACK
> | abort_stmt    // EXTEND WITH HELP: ABORT
> ```

I'd recommend against EXTEND WITH HELP here if you're not also providing `// %Help` below.
Also since this is just syntactic sugar, what about no contextual help at all. 

Also: `/* SKIP DOC */` here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20274#-:-KzyRah9KMvugcmqXcuc:b-xvp6sh)*

**Comment by jordanlewis** [comment]
Done. PTAL.

File: pkg/sql/vars.go [hunk 1]
```diff
@@ -229,6 +229,25 @@ var varGen = map[string]sessionVar{
 	// See https://www.postgresql.org/docs/9.6/static/runtime-config-client.html
 	`extra_float_digits`: nopVar,
 
+	// Supported for PG compatibility only.
+	// See https://www.postgresql.org/docs/9.4/static/runtime-config-client.html
```
Comment (knz) [inline pkg/sql/vars.go]: Not worth a CI run, but next time please grab the most recent version of the doc link.

**Review by knz — APPROVED** [review]

**Comment by jordanlewis** [comment]
Review status: 1 of 8 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/vars.go, line 233 at r4](https://reviewable.io:443/reviews/cockroachdb/cockroach/20274#-KzyQSwBpJu4szwH7Khf-r4-233:-KzzEpb-0uTpNZtTutZD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d21c7cbe8168e6a4e14c0348341b399408854d3d/pkg/sql/vars.go#L233)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Not worth a CI run, but next time please grab the most recent version of the doc link.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/20274)*

**Pull request merged**